test/unit/Reflection/ReflectionFunctionAbstractTest.php

```diff
@@ -236,6 +236,9 @@ public static function generatorProvider(): array
             ['<?php function foo() { $foo->func(yield $foo); }', true],
             ['<?php function foo() { new Foo(yield $foo); }', true],
             ['<?php function foo() { yield from []; }', true],
+            ['<?php function foo() { return fn () => yield $foo; }', false],
+            ['<?php function foo() { return function () { yield $foo; }; }', false],
+            ['<?php function foo() { return new class () { function bar() { yield $foo; } }; }', false]
```

Comment:
I think we even had a PR discussion about this, last year: thanks for fixing this pothole!